src/pypdfium2/internal/utils.py

```diff
@@ -45,9 +45,9 @@ def __init__(self, buffer):
         self.buffer = buffer
     
     def __call__(self, _, position, p_buf_first, size):
-        p_buf = ctypes.cast(p_buf_first, ctypes.POINTER(ctypes.c_char * size))
+        buf = (ctypes.c_char * size).from_address( ctypes.addressof(p_buf_first.contents) )
```

Comment:
Fun fact: This is actually a regression introduced with https://github.com/pypdfium2-team/pypdfium2/commit/6907ddf46ea21562c8b3a312be8f1e00b148b9f1. Historically, we had done it with `from_address()` already.

(See https://github.com/pypdfium2-team/pypdfium2/blame/cc0c2b619b0f918995d1b34aa4c91808de047262/src/pypdfium2/_helpers/_opener.py and click "Blame prior to change" on the line in question.)

Reply:
In terms of versions, it was right until `2.11.0 (2022-08-22)`, and regressed with `3.0.0b1 (2022-09-24)`.